driver-core/src/main/java/com/datastax/driver/core/ParseUtils.java

```diff
@@ -428,45 +460,126 @@ public static String formatTime(long value) {
+        // don't handle non quote characters, this is done so that these are interned and don't create
+        // repeated empty quoted strings.
+        assert quoteChar == '"' || quoteChar == '\'';
+        if (quoteChar == '"')
+            return "\"\"";
+        else
+            return "''";
+    }
+
+
+    /**
+     * Quotes text and escapes any existing quotes in the text.
+     * {@code String.replace()} is a bit too inefficient (see JAVA-67, JAVA-1262).
+     *
+     * @param text      The text.
+     * @param quoteChar The character to use as a quote.
+     * @return The text with surrounded in quotes with all existing quotes escaped with (i.e. ' becomes '')
+     */
+    private static String quote(String text, char quoteChar) {
         if (text == null || text.isEmpty())
-            return "";
+            return emptyQuoted(quoteChar);
 
         int nbMatch = 0;
         int start = -1;
         do {
-            start = text.indexOf(search, start + 1);
+            start = text.indexOf(quoteChar, start + 1);
             if (start != -1)
                 ++nbMatch;
         } while (start != -1);
 
+        // no quotes found that need to be escaped, simply surround in quotes and return.
         if (nbMatch == 0)
+            return quoteChar + text + quoteChar;
+
+        // 2 for beginning and end quotes.
+        // length for original text
+        // nbMatch for escape characters to add to quotes to be escaped.
+        int newLength = 2 + text.length() + nbMatch;
+        char[] result = new char[newLength];
+        result[0] = quoteChar;
+        result[newLength - 1] = quoteChar;
+        int newIdx = 1;
+        for (int i = 0; i < text.length(); i++) {
+            char c = text.charAt(i);
+            if (c == quoteChar) {
+                // escape quote with another occurrence.
+                result[newIdx++] = c;
+                result[newIdx++] = c;
+            } else {
+                result[newIdx++] = c;
+            }
+        }
+        return new String(result);
+    }
+
+    /**
+     * Unquotes text and unescapes non surrounding quotes.
+     * {@code String.replace()} is a bit too inefficient (see JAVA-67, JAVA-1262).
+     *
+     * @param text      The text
+     * @param quoteChar The character to use as a quote.
+     * @return The text with surrounding quotes removed and non surrounding quotes unescaped (i.e. '' becomes ')
+     */
+    private static String unquote(String text, char quoteChar) {
+        if (!isQuoted(text, quoteChar))
             return text;
 
-        int newLength = text.length() + nbMatch * (replacement.length() - 1);
+        String search = emptyQuoted(quoteChar);
+        int nbMatch = 0;
+        int start = -2;
+        do {
+            start = text.indexOf(search, start + 2);
+            // ignore the second to last character occurrence, as the last character is a quote.
+            if (start != -1 && start != text.length() - 2)
+                ++nbMatch;
```

Comment:
> `start != text.length() - 2`

Is the goal here to handle malformed strings that end with a double quote, as in `ParseUtils.unquote("'foo''")`? It looks like it also covers the quoted empty string `"''"`, however it fails on `"'''"` (three consecutive quotes). This can be solved by initializing `start` at -1 instead of -2.